.github/workflows/js.yml

```diff
@@ -0,0 +1,102 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: NodeJS
+
+on:
+  push:
+  #   paths:
+  #     - 'js/**'
+  pull_request:
+  #   paths:
+  #     - 'js/**'
+
+jobs:
+  docker:
+    name: AMD64 Debian NodeJS ${{ matrix.node }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        node: [11]
+    env:
+      NODE: ${{ matrix.node }}
+    steps:
+      - name: Checkout Arrow
+        uses: actions/checkout@v1
+        with:
+          submodules: true
+      - name: Docker Pull
+        run: docker-compose pull --ignore-pull-failures debian-js
+      - name: Docker Build
+        run: docker-compose build debian-js
+      - name: Docker Run
+        run: docker-compose run debian-js
+      - name: Docker Push
+        if: github.event_name == 'push' # && github.ref == 'master'
+        shell: bash
+        run: |
+          docker login \
+            -u ${{ secrets.DOCKERHUB_USER }} \
+            -p ${{ secrets.DOCKERHUB_TOKEN }}
+          docker-compose push debian-js
+
+  macos:
+    name: AMD64 MacOS 10.15 NodeJS ${{ matrix.node }}
+    runs-on: macos-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        node: [11]
+    steps:
+      - name: Checkout Arrow
+        uses: actions/checkout@v1
+        with:
+          submodules: true
+      - name: Install NodeJS
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node }}
+      - name: Build
+        shell: bash
+        run: ci/scripts/js_build.sh $(pwd)
+      - name: Test
+        shell: bash
+        run: ci/scripts/js_test.sh $(pwd)
+
+  windows:
+    name: AMD64 Windows 2019 NodeJS ${{ matrix.node }}
+    runs-on: windows-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        node: [11]
+    steps:
+      - name: Checkout Arrow
+        uses: actions/checkout@v1
+        with:
+          submodules: true
+      - name: Install NodeJS
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node }}
+      - name: Build
```

Comment:
@trxcllnt not sure why does it [fail](https://github.com/apache/arrow/pull/5589/checks?check_run_id=288536561), with an earlier windows worker it was working. Do I need to nuke node_modules update npm?

Reply:
Oh Windows. We'll need to use [cross-env](https://www.npmjs.com/package/cross-env) to set that environment variable.